backend/models/helper/matrix.js

```diff
@@ -0,0 +1,110 @@
+/**
+ *	Copyright (C) 2020 3D Repo Ltd
+ *
+ *	This program is free software: you can redistribute it and/or modify
+ *	it under the terms of the GNU Affero General Public License as
+ *	published by the Free Software Foundation, either version 3 of the
+ *	License, or (at your option) any later version.
+ *
+ *	This program is distributed in the hope that it will be useful,
+ *	but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *	GNU Affero General Public License for more details.
+ *
+ *	You should have received a copy of the GNU Affero General Public License
+ *	along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+"use strict";
+
+const multiply = (matrixA, matrixB) => {
+	const res = new Array(matrixA.length);
+
+	for (let i = 0; i < matrixA.length ; i++) {
+		const row = matrixA[i];
+		const newRow = new Array(matrixA.length);
+		res[i] = newRow;
+		for (let j = 0; j < matrixA.length; j++) {
+			newRow[j] = 0;
+			for (let h = 0; h < matrixA.length; h++) {
+				newRow[j] += row[h] * matrixB[h][j];
+			}
+
+		}
+	}
+
+	return res;
+};
+
+const getIdentity = (size) => {
+	const matrix =  new Array(size);
+	for (let i = 0 ; i < size; i++) {
+		matrix[i] = new Array(size);
+		for (let j = 0 ; j < size; j++) {
+			matrix[i][j] = i !== j ? 0 : 1;
+		}
+	}
+
+	return matrix;
+};
+
+const scalarMultiply = (matrix, scalar) => {
```

Comment:
@sanmont3drepo 
This doesn't look like it's used?

I don't think you'd ever multiply a matrix with a scalar like this.

If the purpose is to scale a vector it's only applied on numbers where i=j (apart from the last line).

Reply:
Yes, TBH I dont know why I added that.